application/CMakeLists.txt

```diff
@@ -56,11 +56,20 @@ elseif (MSVC)
   set_target_properties(f3d PROPERTIES VS_DEBUGGER_ENVIRONMENT "PATH=$<TARGET_FILE_DIR:VTK::CommonCore>") # Add VTK in MSVC PATH environment variable
 endif ()
 
-target_include_directories(f3d
-  PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/external>
-  )
+target_include_directories(f3d PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
+if (F3D_USE_EXTERNAL_CXXOPTS)
+  find_package(cxxopts REQUIRED)
+  target_link_libraries(f3d PRIVATE cxxopts::cxxopts)
+else ()
+  target_include_directories(f3d PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/external/cxxopts>)
+endif ()
+
+if (F3D_USE_EXTERNAL_NLOHMANN_JSON)
+  find_package(nlohmann_json REQUIRED)
+  target_link_libraries(f3d PRIVATE nlohmann_json::nlohmann_json)
+else ()
+  target_include_directories(f3d PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/external>)
```

Comment:
I dont like that because if we add another external, we may include the wrond header. Please put `nlohmann` dir in a dedicated directory.

Reply:
https://github.com/f3d-app/f3d/actions/runs/5120469346/jobs/9207007100?pr=832

Reply:
I am a little confused about the problem of including the header file.

Reply:
looking at the code it should work, So I do not know. Are you able to reproduce locally ?

Reply:
nlohmann_json is used in libf3d, not F3D app, see here: https://github.com/f3d-app/f3d/blob/614264de76967461cca02fed3daccb3548b8c5c7/library/CMakeLists.txt#L102

Reply:
Actually, it's used in both. So you also have to change it in libf3d but keep the changes in F3D app.

Reply:
Thanks for pointing out the problem.